compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -614,6 +614,102 @@ static LogicalResult setTransposeConfig(func::FuncOp entryPoint,
       workgroupSize);
 }
 
+static LogicalResult setConvolutionConfig(linalg::LinalgOp linalgOp,
+                                          const int64_t subgroupSize,
+                                          const int64_t bestTilingFactor) {
+  Type inputType = linalgOp.getInputOperand(0)->get().getType();
+  ArrayRef<int64_t> inputShape = inputType.cast<ShapedType>().getShape();
+  Type outputType = linalgOp.getOutputOperand(0)->get().getType();
+  ArrayRef<int64_t> outputShape = outputType.cast<ShapedType>().getShape();
+  if (!isa<linalg::Conv2DNhwcHwcfOp>(*linalgOp) ||
+      ShapedType::isDynamic(inputShape[3])) {
+    return failure();
+  }
+  if (llvm::any_of(outputShape.drop_front(), ShapedType::isDynamic)) {
+    return failure();
+  }
+  int64_t ic = inputShape[3];
+  int64_t oh = outputShape[1], ow = outputShape[2], oc = outputShape[3];
+  // The conversion pipeline requires the input channel dimension to be some
+  // multipler of four, or less than four.
+  if (!(ic % 4 == 0 || ic < 4)) return failure();
```

Comment:
This check is actually true for the IREE version of convolution vectorization. Not really for upstream one. So it can be dropped.

Reply:
Thanks, removed